Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.CssEditor/scripts/CssEditor.js

```diff
@@ -97,26 +98,82 @@ define([
             if (params.settings.isHost) {
                 getPortals();
             }
-            getStyleSheet();
+            
+            initCssEditor();
 
-            cssEditor = codeEditor.init($panel.find('textarea'), { mode: 'css' });
             var panelHeight = $('#CssEditor-panel').height();
             if (panelHeight > 400) {
-                cssEditor.setSize("100%", panelHeight - 330);
+                $('#monaco-editor').height(panelHeight - 400);
             }
 
             if (typeof callback === 'function') {
                 callback();
             }
 
-            cssEditor.on("blur", function(cm) {
+            $('#monaco-editor').on("blur", function(cm) {
                 cm.save();
                 return true;
             });
 
             viewModel.portal.subscribe(portalChanged);
         };
 
+        var initCssEditor = function() {
+            var monacoEditorLoaderScript = document.createElement('script');
+            monacoEditorLoaderScript.type = 'text/javascript';
+            monacoEditorLoaderScript.src = '/Resources/Shared/components/MonacoEditor/loader.js';
+            document.body.appendChild(monacoEditorLoaderScript);
+
+            require.config({ paths: { 'vs': '/Resources/Shared/components/MonacoEditor' }});
```

Comment:
Using requireJS is a supported code path for MonacoEditor and it is a great way to get started. The DNN PersonaBar core modules use ReactJS and the MonacoEditor has a ReactJS component that is easy to use in that context. If we are adding this control to a ReactJS page I would consider using that component.

I am not 100% sure how this particular module uses the ReactJS components that is the DNN persona bar modules. In the context of ReactJS I find it much easier to configure the editor and manage it than calling the JS apis. That is my preference and I defer to the maintainers opinions on this one.

Reply:
This Persona Bar extension (CSS Editor) does not use React.  I'm not sure if the other extensions which might use Monaco use React or not (SQL and Config Manager).

Reply:
@ahoefling this module currently is not using ReactJS.  It is plain HTML, JS, CSS.  Therefore, this is a stepping stone solution as stated in #4132.

Reply:
Furthermore, I would love that we don't make it React only :) But agree we should pull it as an npm package, which is a future step after this stepping stone.